homework/index.js

```diff
@@ -30,15 +35,97 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderError(err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  }
+
+  // deletes the previous repository table if selected repository changes
+  function removeChildElements(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+
+  // to show the information of selected repository in a table
+  function showRepoInfo(index, repoDiv, repoArray, containerDiv, contributorsDiv) {
+    removeChildElements(repoDiv);
+    const selectedRepo = repoArray[index];
+    const headerArray = ['Repository', 'Description', 'Forks', 'Updated'];
+    const repoInfo = [
+      selectedRepo.name,
+      selectedRepo.description,
+      selectedRepo.forks,
+      selectedRepo.updated_at,
+    ];
+    const table = createAndAppend('table', repoDiv);
+    for (let i = 0; i < headerArray.length; i++) {
+      const tr = createAndAppend('tr', table);
+      createAndAppend('td', tr, { text: headerArray[i], class: 'label' });
+      if (i === 0) {
+        const repoLink = createAndAppend('td', tr);
+        createAndAppend('a', repoLink, {
+          href: selectedRepo.svn_url,
```

Comment:
Use `html_url` attribute instead of `svn_url`, even though here they are resulting to the same value, but they have different meanings. See [SVN support](https://help.github.com/en/articles/support-for-subversion-clients)

Reply:
Also fixed.